dcr/walletdata.go

```diff
@@ -1,43 +1,52 @@
-package asset
+package dcr
 
 import (
 	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 	"time"
 )
 
 const walletDataFileName = "walletdata.json"
 
 type walletData struct {
+	// seedMtx protects the EncryptedSeedHex field which may be modified when the
+	// wallet password is changed.
+	seedMtx            sync.Mutex
```

Comment:
This mutex makes more sense as part of the `Wallet` struct to me because the methods that use it are there.